hawc/apps/riskofbias/serializers.py

```diff
@@ -125,18 +130,127 @@ class Meta:
+            # (2) for scores, a default=True exists for each required metric (by study type) for a study.
+
+            # (1) does the study already have an active=True/final=True rob?
+            if self.initial_data["active"] is True and self.initial_data["final"] is True:
+                robs = study.riskofbiases.all()
+                for rob in robs:
+                    if rob.active and rob.final:
+                        raise serializers.ValidationError(
+                            "create failed; study %s already has an active & final risk of bias"
+                            % study_id
+                        )
+
+            # (2) subject to the settings of the metrics defined for this assessment,
+            # and the study_type of the study, we need to make sure all necessary scores
+            # have been submitted with a default=True
+            scores = self.initial_data["scores"]
+            required_metrics = models.RiskOfBiasMetric.objects.get_required_metrics(
+                assessment, study
+            )
+            problematic_scores = []
+            for metric in required_metrics:
+                domain = metric.domain
+                # there could be multiple scores for a given metric if there are overrides, so we need to fetch them all
+                scores_for_metric = [
+                    score
+                    for score in scores
+                    if "metric_id" in score and score["metric_id"] == metric.id
+                ]
+
+                metric_descriptor = "'%s:%s'" % (domain.name, metric.name)
+
+                if 0 == len(scores_for_metric):
+                    problematic_scores.append(
+                        "No score for metric %s/%s was submitted" % (metric.id, metric_descriptor)
+                    )
+                else:
+                    default_score_for_this_metric_included = False
+                    for submitted_score in scores_for_metric:
+                        if (
+                            "is_default" in submitted_score
+                            and submitted_score["is_default"] is True
+                        ):
+                            default_score_for_this_metric_included = True
+                            break
+
+                    if not default_score_for_this_metric_included:
+                        problematic_scores.append(
+                            "The score for metric %s/%s was submitted but not marked as default"
+                            % (metric.id, metric_descriptor)
+                        )
+
+            if len(problematic_scores) > 0:
+                explanation = "; ".join(problematic_scores)
+                raise serializers.ValidationError(
+                    "create failed; study %s had problematic scores (%s)" % (study_id, explanation)
+                )
+
+            # store the actual metric object we want to create
+            score_idx = 0
```

Comment:
do this instead; fewer queries and cleaner:
```python
# store the actual metric object we want to create
metrics_dict = {obj.id: obj for obj in required_metrics}
for score, initial_score in zip(data["scores"], scores):
    score["metric"] = metrics_dict[initial_score["metric_id"]]
```

Reply:
ah ok, I thought I needed the entire metric object in there to make it work. fixed and will be in next push.